document virtualColumns in native query documentation, fix some redirects (#12917)
apache/druid: docs/querying/groupbyquery.md

|queryType|This String should always be "groupBy"; this is the first thing Druid looks at to figure out how to interpret the query|yes|
|dataSource|A String or Object defining the data source to query, very similar to a table in a relational database. See [DataSource](../querying/datasource.md) for more information.|yes|
|dimensions|A JSON list of dimensions to do the groupBy over; or see [DimensionSpec](../querying/dimensionspecs.md) for ways to extract dimensions. |yes|

> |virtualColumns|Define any [virtual columns](./virtual-columns.md) to be available to use as `dimensions` or referenced as inputs to `aggregations` or `postAggregations`.| no (default none)|

Reviewer: ```suggestion
|virtualColumns|A JSON list of [virtual columns](./virtual-columns.md). You can reference the virtual columns in `dimensions`, `aggregations`, or `postAggregations`.| no (default none)|
```
